yogstation/code/modules/clothing/suits/nerd.dm

```diff
@@ -0,0 +1,217 @@
+	icon = 'yogstation/icons/obj/clothing/suits.dmi'
+	icon_state = "nerd"
+	item_state = "nerd"
+	blood_overlay_type = "suit"
+	resistance_flags = FIRE_PROOF | ACID_PROOF
+	armor = list(MELEE = 25, BULLET = 25, LASER = 25, ENERGY = 25, BOMB = 0, BIO = 100, RAD = 100, FIRE = 50, ACID = 50)
+	allowed = list(/obj/item/tank/internals/emergency_oxygen, /obj/item/tank/internals/plasmaman, /obj/item/gun/energy/kinetic_accelerator, /obj/item/tank/internals/ipc_coolant, /obj/item/crowbar)
+
+	var/list/funny_signals = list(
+		COMSIG_MOB_SAY = .proc/handle_speech,
+		COMSIG_MOB_DEATH = .proc/handle_death,
+		COMSIG_LIVING_IGNITED = .proc/handle_ignite,
+		COMSIG_LIVING_ELECTROCUTE_ACT = .proc/handle_shock,
+		COMSIG_CARBON_GAIN_WOUND = .proc/handle_wound_add,
+		COMSIG_MOB_APPLY_DAMAGE = .proc/handle_damage
+	)
+
+	var/list/wound_to_sound = list(
+		/datum/wound/blunt/severe = 'sound/voice/nerdsuit/minor_fracture.ogg',
+		/datum/wound/blunt/critical = 'sound/voice/nerdsuit/major_fracture.ogg',
+		/datum/wound/loss = 'sound/voice/nerdsuit/major_lacerations.ogg',
+		/datum/wound/slash/severe = 'sound/voice/nerdsuit/minor_lacerations.ogg',
+		/datum/wound/slash/critical = 'sound/voice/nerdsuit/major_lacerations.ogg',
+	)
+
+	var/list/sound_queue = list()
+
+	var/damage_notify_next = 0
+	var/morphine_next = 0
+	var/sound_next = 0
+
+	var/mob/living/carbon/owner
+
+	var/emagged = FALSE
+
+	body_parts_covered = CHEST|GROIN|ARMS|LEGS|FEET
+	flags_inv = HIDESHOES|HIDEJUMPSUIT|HIDEGLOVES
+
+/obj/item/clothing/suit/armor/nerd/Initialize()
+	. = ..()
+	var/obj/item/geiger_counter/GC = new(src)
+	GC.scanning = TRUE
+	update_icon()
+
+/obj/item/clothing/suit/armor/nerd/proc/process_sound_queue()
+
+	if(sound_next > world.time)
+		return FALSE
+
+	if(!length(sound_queue))
+		return FALSE
+
+	var/list/sound_data = sound_queue[1]
+	var/sound_file = sound_data[1]
+	var/sound_delay = sound_data[2]
+
+	playsound(src, sound_file, 50)
+	sound_next = world.time + sound_delay
+
+	sound_queue.Cut(1,2)
+
+	addtimer(CALLBACK(src, .proc/process_sound_queue), sound_delay)
+
+/obj/item/clothing/suit/armor/nerd/emag_act(mob/user)
+
+	if(owner)
+		return FALSE //Take it off!
+
+	if(emagged == FALSE)
+		emagged = TRUE
+		do_sparks(8, FALSE, get_turf(src))
+		return TRUE
+	return FALSE
+
+/obj/item/clothing/suit/armor/nerd/proc/add_queue(var/desired_file,var/desired_delay,var/purge_queue=FALSE)
+
+	var/empty_sound_queue = !length(sound_queue)
+
+	if(purge_queue)
+		sound_queue.Cut()
+
+	sound_queue += list(list(desired_file,desired_delay)) //BYOND is fucking weird so you have to do this bullshit if you want to add a list to a list.
+
+	if(empty_sound_queue)
+		addtimer(CALLBACK(src, .proc/process_sound_queue), 10)
+
+	return TRUE
+
+
+//Signal handling.
+/obj/item/clothing/suit/armor/nerd/equipped(mob/M, slot)
+	. = ..()
+	if(slot == SLOT_WEAR_SUIT && iscarbon(M))
+		for(var/k in funny_signals)
+			RegisterSignal(M, k, funny_signals[k])
```

Comment:
Ensure each signal handler proc has the ``SIGNAL_HANDLER`` define at the beginning